src/features/game/expansion/placeable/lib/collisionDetection.ts

```diff
@@ -0,0 +1,205 @@
+import omit from "lodash.omit";
+import { GameState, LandExpansion, Position } from "features/game/types/game";
+import { EXPANSION_ORIGINS, LAND_SIZE } from "../../lib/constants";
+import { Coordinates } from "../../components/MapPlacement";
+
+/**
+ * Extracts positional data for all instances of a single resource
+ * @param resource
+ * @param expansionIndex
+ * @returns Array containaing all positional data for one resource type e.g. Shrub
+ */
+const extract = <T extends Record<number, Position>>(
+  resource: T,
+  expansionIndex: number
+): Position[] => {
+  const { x: xOffset, y: yOffset } = EXPANSION_ORIGINS[expansionIndex];
+
+  return Object.values(resource).map(({ x, y, height, width }) => ({
+    x: x + xOffset,
+    y: y + yOffset,
+    height,
+    width,
+  }));
+};
+
+/**
+ * Extracts the positional data from all the resources on all land expansions.
+ * @param expansions
+ * @returns Array of all resource positions
+ */
+export function extractResourcePositions(
+  expansions: Required<
+    Omit<LandExpansion, "terrains" | "createdAt" | "readyAt">
+  >[]
+) {
+  return expansions.flatMap((expansion, expansionIndex) =>
```

Comment:
Can we break this apart. It is a bit hard to follow, I'm not sure what is going on here 😅 

`flatMap` -> `Object.values` -> `omit` -> `flatMap` -> `extract`